### PR TITLE
connectors-ci: send slack message and set execute timeout

### LINKED ISSUE #25844: connectors-ci: send slack messages for publish pipelines
When a publish pipeline is triggered on merge to master we want to receive slack messages in a channel:
* When a connector specific publish pipeline started
* When it fails (tag @dev-connector-ops)
* When it succeeds

### LINKED ISSUE #25845: connectors-ci: publish pipeline should timeout after 1H
We want dagger operations to fail after 1H timeout on publish pipelines

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes #25844
Closes #25845

We want to send update to slack channels when a pipeline context starts and stops (success or failure).
We also want publish pipeline to fail if a dagger operation takes more than 1H.

## How
* Send message to #publish-on-merge-updates when entering or exiting 
* Set an execute timeout to 1H
Screen Shot 2023-05-05 at 16 33 58

### CODE REVIEW

File: tools/ci_connector_ops/ci_connector_ops/pipelines/commands/groups/connectors.py [hunk 1]
```diff
@@ -77,6 +77,12 @@ def validate_environment(is_local: bool, use_remote_secrets: bool):
 )
 @click.option("--modified/--not-modified", help="Only test modified connectors in the current branch.", default=False, type=bool)
 @click.option("--concurrency", help="Number of connector tests pipeline to run in parallel.", default=5, type=int)
+@click.option(
+    "--execute-timeout",
+    help="The maximum time in seconds for the execution of a Dagger request before an ExecuteTimeoutError is raised. Passing None results in waiting forever.",
+    default=None,
```
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/commands/groups/connectors.py:83]: Is a 1h default not good for all types of tasks (except maybe the nightly tests?)

**Review by evantahler — APPROVED** [review]
Nice!

Nit: In the screenshot the title of the message is `<none|publish>`... what's `none`?

File: tools/ci_connector_ops/ci_connector_ops/pipelines/commands/groups/connectors.py (hunk 1 above)
Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/pipelines/commands/groups/connectors.py:83]: I wanted to make this specific to publish as until now I haven't been setting timeouts. It should indeed be good for all types of task but to reduce moving parts I prefer to not set the timeout globally.

**Comment by alafanechere** [comment]
> Nit: In the screenshot the title of the message is <none|publish>... what's none?

none should be the link to the GHA workflow. As I ran the command locally for testing the GHA workflow URL is none.

**Pull request merged**